tests/scripts/all.sh

```diff
@@ -5117,6 +5117,16 @@ support_build_cmake_custom_config_file () {
 }
 
 
+component_build_zeroize_checks () {
+    msg "build: check for obviously wrong calls to mbedtls_platform_zeroize()"
```

Comment:
A side note, we've considered running clang-tidy and its [`bugprone-sizeof-expression`](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/sizeof-expression.html) looks for the same kind of problem. I don't know whether it would catch invalid calls in a normal build or if we'd need this memset replacement.

I thought we had an open issue for clang-tidy but turns out we don't, I'll file one (as part of the huge tech debt backlog, so don't hold your breath).